actions/rules/action_state_to_metastate.go

```diff
@@ -0,0 +1,459 @@
+			}
+			columnsAdded, _ = act.difference(oldValue, newValue)
+		}
+	}
+	// from here on, we ignore the removed columns as a possible metastate transition is
+	// only happening on *adding* a new column. Removing column does not trigger a change.
+	if len(columnsAdded) == 0 {
+		// somehow, no actual changes on the columns.
+		return newContext, *actionChanges, nil
+	}
+	// get the mapping.
+	mapping, err := act.getMetastateToStateMap(wi.Type)
+	if err != nil {
+		return nil, nil, err
+	}
+	// create a dirty flag. Note that we can not use len(actionChanges) as this
+	// may contain previous changes from the action chain.
+	wiDirty := false
+	var changes change.Set
+	// go over all added columns. We support multiple added columns at once.
+	for _, columnID := range columnsAdded {
+		// get board and column by columnID.
+		var thisColumn workitem.BoardColumn
+		boards, err := act.loadWorkItemBoardsBySpaceID(wi.SpaceID)
+		if err != nil {
+			return nil, nil, err
+		}
+		for _, board := range boards {
+			for _, column := range board.Columns {
+				if columnID == column.ID.String() {
+					thisColumn = column
+				}
+			}
+		}
+		// at this point, we don't check if the board was
+		// relevant (matching the type group) as the move into
+		// the column has already happened. We just make sure
+		// this rule applies to the column.
+		if thisColumn.TransRuleKey != ActionKeyStateToMetastate {
+			// this is a column that does not apply to the rule, we don't apply here.
+			return newContext, *actionChanges, nil
+		}
+		// unmarshall the configuration.
+		config := map[string]string{}
+		err = json.Unmarshal([]byte(thisColumn.TransRuleArgument), &config)
+		if err != nil {
+			return nil, nil, err
+		}
+		// extract the column's metastate config.
+		if metaState, ok := config[ActionKeyStateToMetastateConfigMetastate]; ok {
+			if metaState == wi.Fields[workitem.SystemMetaState] {
+				// the WIs metastate is already the same as the columns
+				// metastate, so nothing to do.
+				return newContext, *actionChanges, nil
+			}
+			// the metatstate changes, so set it on the WI.
+			changes = act.addOrUpdateChange(changes, workitem.SystemMetaState, wi.Fields[workitem.SystemMetaState], metaState)
+			wi.Fields[workitem.SystemMetaState] = metaState
+			wiDirty = true
+			// next, check if the state needs to change as well from the metastate.
+			if wi.Fields[workitem.SystemState] != mapping[metaState] {
+				// yes, the state changes as well.
+				changes = act.addOrUpdateChange(changes, workitem.SystemState, wi.Fields[workitem.SystemState], mapping[metaState])
+				wi.Fields[workitem.SystemState] = mapping[metaState]
+			}
+		}
+	}
+	// finally, store the new work item state if something changed.
+	var resultingWorkItem workitem.WorkItem
+	if wiDirty {
+		result, err := act.storeWorkItem(&wi)
+		resultingWorkItem = *result
+		if err != nil {
+			return nil, nil, err
+		}
+	}
+	// return to sender
+	return resultingWorkItem, changes, nil
```

Comment:
@michaelkleinhenz We're returning an empty workitem here if `wiDirty` is false. This isn't expected, right? 
Can you please fix it?

Reply:
Yeah. Fixed. 6f4d7fb.

Reply:
Removed the redundant var in 3cb93a2.